### PR TITLE
feat: user clients verification

### PR DESCRIPTION
**Pull request by MohamadJaara**
----
#### PR Submission Checklist for internal contributors

- The **PR Title**
    - [ ] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
    - [ ] contains a reference JIRA issue number like `SQPIT-764`
    - [ ] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
    - [ ] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues

1. self list of clients are always fetched from remote even if the info is locally available.
2. the registration time is not visible for self clients.

### Solutions

1. implement clients verification.
2. update the self list of clients from remote when opening the screen.
3. observe the list of self clients.
4. display clients fingerprint.

### Dependencies (Optional)

_If there are some other pull requests related to this one (e.g. new releases of frameworks), specify them here._

Needs releases with:

- [ ] GitHub link to other pull request

#### Test Coverage (Optional)

- [ ] I have added automated test to this contribution

#### How to Test

_Briefly describe how this change was tested and if applicable the exact steps taken to verify that it works as expected._

### Notes (Optional)

_Specify here any other facts that you think are important for this issue._

### Attachments (Optional)

_Attachments like images, videos, etc. (drag and drop in the text box)_ 

----
#### PR Post Submission Checklist for internal contributors (Optional)

- [ ] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

- [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

**Comment by AndroidBob** [comment]
Build [1](https://10.10.124.134/job/wire-android-reloaded/job/PR-1552/1/) **failed**.

**Comment by AndroidBob** [comment]
Build [2](https://10.10.124.134/job/wire-android-reloaded/job/PR-1552/2/) **failed**.

**Comment by AndroidBob** [comment]
Build [3](https://10.10.124.134/job/wire-android-reloaded/job/PR-1552/3/) **failed**.

**Comment by AndroidBob** [comment]
Build [5](https://10.10.124.134/job/wire-android-reloaded/job/PR-1552/5/) **failed**.

**Comment by AndroidBob** [comment]
Build [6](https://10.10.124.134/job/wire-android-reloaded/job/PR-1552/6/) **failed**.

**Comment by AndroidBob** [comment]
Build [8](https://10.10.124.134/job/wire-android-reloaded/job/PR-1552/8/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv4.0.2(6008468)(6008468)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-vv4.0.2(6008468)(6008468)-staging-compat.apk)

**Comment by AndroidBob** [comment]
Build [10](https://10.10.124.134/job/wire-android-reloaded/job/PR-1552/10/) **failed**.

**Comment by AndroidBob** [comment]
Build [11](https://10.10.124.134/job/wire-android-reloaded/job/PR-1552/11/) **failed**.

**Comment by AndroidBob** [comment]
Build [12](https://10.10.124.134/job/wire-android-reloaded/job/PR-1552/12/) **failed**.

**Comment by AndroidBob** [comment]
Build [13](https://10.10.124.134/job/wire-android-reloaded/job/PR-1552/13/) **failed**.

**Comment by AndroidBob** [comment]
Build [14](https://10.10.124.134/job/wire-android-reloaded/job/PR-1552/14/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv4.0.2(6034719)(6034719)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-vv4.0.2(6034719)(6034719)-staging-compat.apk)

**Review by borichellow — APPROVED** [review]
👍

**Comment by AndroidBob** [comment]
Build [16](https://10.10.124.134/job/wire-android-reloaded/job/PR-1552/16/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv4.0.2(6035060)(6035060)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-vv4.0.2(6035060)(6035060)-staging-compat.apk)

**Comment by AndroidBob** [comment]
Build [17](https://10.10.124.134/job/wire-android-reloaded/job/PR-1552/17/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv4.0.2(6036189)(6036189)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-vv4.0.2(6036189)(6036189)-staging-compat.apk)

File: app/src/main/kotlin/com/wire/android/ui/settings/devices/DeviceDetailsScreen.kt [hunk 1]
```diff
@@ -162,12 +206,171 @@ fun DeviceDetailsContent(
...
+                start = dimensions().spacing16x,
+                end = dimensions().spacing16x,
+                bottom = dimensions().spacing16x
+            )
+    ) {
+
+        if (isSelfClient) {
+            DescriptionText(
+                text = stringResource(id = R.string.label_self_client_verification_description),
+                leanMoreLink = "https://support.wire.com/hc/en-us/articles/207692235-How-can-I-compare-key-fingerprints-"
```
Comment (gongracr) [inline app/src/main/kotlin/com/wire/android/ui/settings/devices/DeviceDetailsScreen.kt]: Let's localize all these support links to the strings file, similar to what we did to others. There is probably a German support version of them, so ze German users will appreciate it too 🤓

File: app/src/main/res/values/strings.xml [hunk 2]
```diff
@@ -917,6 +917,10 @@
     <string name="guest_link_revoke_error_message">There was an error trying to revoke Guest Link. Please try again</string>
     <string name="revoke_guest__room_link_dialog_text">New guests will not be able to join with this link. Current guests will still have access.</string>
     <string name="revoke_guest__room_link_dialog_title">Revoke guest link?</string>
+    <string name="guest__room_link_copied">Guest link was copied to clipboard</string>
+    <string name="title_device_key_fingerprint">DEVICE KEY FINGERPRINT</string>
+    <string name="label_client_key_fingerprint_not_available">NOT AVILABLE</string>
```
Comment (gongracr) [inline app/src/main/res/values/strings.xml]:
```suggestion
    <string name="label_client_key_fingerprint_not_available">NOT AVAILABLE</string>
```

File: app/src/main/res/values/strings.xml [hunk 3]
```diff
@@ -928,4 +932,11 @@
     <string name="new_device_dialog_other_user_title">Your account “%1$s (@%2$s)” was used on:</string>
     <string name="new_device_dialog_other_user_message">%1$s\nFrom: %2$s \n\nIf you didn’t do this, remove the device (navigate to “Manage Devices” in the Settings section of this account), and reset your password.</string>
     <string name="new_device_dialog_other_user_btn">Switch Account</string>
+    <string name="label_client_verified">Verified</string>
+    <string name="label_client_unverified">Not Verified</string>
+    <string name="label_self_client_verification_description">To verify your own device, compare this key  fingerprint with the same key fingerprint on another device.</string>
```
Comment (gongracr) [inline app/src/main/res/values/strings.xml]:
```suggestion
    <string name="label_self_client_verification_description">To verify your own device, compare this key fingerprint with the same key fingerprint on another device.</string>
```

**Review by gongracr — APPROVED** [review]
Just some small strings comments to localise them better. But since they are quite small I'm approving in advance. 
Great job with this feature! 🦾

**Comment by AndroidBob** [comment]
Build [18](https://10.10.124.134/job/wire-android-reloaded/job/PR-1552/18/) **failed**.

**Comment by AndroidBob** [comment]
Build [19](https://10.10.124.134/job/wire-android-reloaded/job/PR-1552/19/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv4.0.2(6044492)(6044492)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-vv4.0.2(6044492)(6044492)-staging-compat.apk)

**Comment by AndroidBob** [comment]
Build [21](https://10.10.124.134/job/wire-android-reloaded/job/PR-1552/21/) **failed**.

**Comment by AndroidBob** [comment]
Build [23](https://10.10.124.134/job/wire-android-reloaded/job/PR-1552/23/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv4.0.2(6045278)(6045278)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-vv4.0.2(6045278)(6045278)-staging-compat.apk)

**Pull request merged**